### ISSUE TITLE
🐛 QueryParser incorrectly parses parameter with multiple values

### ISSUE THREAD

**Issue by Obayanju** [gofiber/fiber#782]
**Fiber version**
v2.x

**Issue description**
QueryParser parses parameters with multiple values as one string instead of multiple strings.

**Code snippet**
https://docs.gofiber.io/ctx#queryparser
```go
// Field names should start with an uppercase letter
type Person struct {
    Name     string     `query:"name"`
    Pass     string     `query:"pass"`
    Products []string   `query:"products"`
}

app.Post("/", func(c *fiber.Ctx) error {
        p := new(Person)

        if err := c.QueryParser(p); err != nil {
            return err
        }

        log.Println(len(p.Products)) // 1 instead of 2 <----------------------

        // ...
})

// Run tests with the following curl command

// curl -X POST "http://localhost:3000/?name=john&pass=doe&products=shoe,hat"
```

**Comment by renanbastos93** [comment]
Hey dude, how ya going?

well, way to use a querystring `?name=john&pass=doe&products=shoe&products=hat`  so it gets an array by products.

**Comment by Obayanju** [comment]
Hello.

I was expecting that `Person.Products` would contain a slice of each of the values of the query parameter `products`. Instead the first value of the slice contains a string of all the values, i.e, shoe, hat

**Comment by renanbastos93** [comment]
another approach you can to make is creating a logic to split by `,` and get your product list.
```go
if len(p.Products) == 1 {
    p.Products = strings.Split(p.Products[0], ",")
    log.Println(len(p.Products), p.Products)
}
```

**Comment by renanbastos93** [comment]
> Hello.
> 
> I was expecting that `Person.Products` would contain a slice of each of the values of the query parameter `products`. Instead the first value of the slice contains a string of all the values, i.e, shoe, hat

@Obayanju I understood, so I hope to help you.

**Comment by Obayanju** [comment]
Thanks you.

**Comment by Fenny** [comment]
Hi, this bug is caused by a bug in the schema library. We opened an issue today https://github.com/gorilla/schema/issues/152.

**Comment by Fenny** [comment]
Quick update, gorilla seems to have no plans to implement this feature so @napei came with the idea to solve it on our end.
See <https://github.com/gofiber/fiber/pull/817>, it will be shipped with the `v2.0.3` tag 👍

**Comment by ugurcanerdogan** [comment]
Is this problem still not solved?
